app/routers/applications.py

```diff
@@ -96,23 +96,30 @@ async def approve_application(
     :return: The approved application with its associated relations.
     """
     with rollback_on_error(session):
-        # Check if all keys present in an instance of UpdateDataField exist and have truthy values in
-        # the application's `secop_data_verification`.
-        app_secop_dict = application.secop_data_verification.copy()
-        fields = list(parsers.UpdateDataField().model_dump().keys())
-        not_validated_fields = [key for key in fields if key not in app_secop_dict or not app_secop_dict[key]]
-        if not_validated_fields:
-            raise HTTPException(
-                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
-                detail=_("Some borrower data field are not verified"),
-            )
+        # If the lender has an external onboarding system the fields and documents verification is not required-
+        if not application.lender.external_onboarding_url:
+            # Check if all keys present in an instance of UpdateDataField exist and have truthy values in
+            # the application's `secop_data_verification`.
+            app_secop_dict = application.secop_data_verification.copy()
+            fields = list(parsers.UpdateDataField().model_dump().keys())
+            not_validated_fields = [key for key in fields if key not in app_secop_dict or not app_secop_dict[key]]
+            if not_validated_fields:
+                raise HTTPException(
+                    status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=_("Some borrower data field are not verified"),
+                )
 
-        # Check all documents are verified.
-        not_validated_documents = [doc.type for doc in application.borrower_documents if not doc.verified]
-        if not_validated_documents:
+            # Check all documents are verified.
+            not_validated_documents = [doc.type for doc in application.borrower_documents if not doc.verified]
+            if not_validated_documents:
+                raise HTTPException(
+                    status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                    detail=_("Some documents are not verified"),
+                )
+        if not payload.disbursed_final_amount:
```

Comment:
You are right, it was left optional because the approval step was also used as "pre-approval" where the amount wasn't required yet. Now, it should be required and the field should be set to Decimal again

Reply:
Mmm actually, after changing it to Decimal only FastAPI doesn't complain if the field is empty and saves it as 0

Reply:
What does the incoming JSON look like? Is the field just blank (empty string) or is it missing?